test/monitor_test.py

```diff
@@ -79,23 +83,43 @@ def f():
         ray.init(redis_address=redis_address)
         # The assertion below can fail if the monitor is too slow to clean up
         # the global state.
-        self.assertEqual((0, 1), StateSummary()[:2])
+
+        if test_object_cleanup:
+            self.assertEqual((0, 1), StateSummary()[:2])
+        else:
+            self.assertEqual(1, StateSummary()[1])
 
         ray.worker.cleanup()
         subprocess.Popen(["ray", "stop"]).wait()
 
+    @unittest.skipIf(
+        os.environ.get('RAY_USE_NEW_GCS', False),
+        "Failing with the new GCS API.")
+    def testTaskCleanupOnDriverExitSingleRedisShard(self):
+        self._testCleanupOnDriverExit(num_redis_shards=1, test_object_cleanup=False)
+
+    @unittest.skipIf(
```

Comment:
I think we should be able to re-enable this.

Reply:
I believe re-enabling this entails a substantial amount of work. It might make sense to re-enable as a separate PR.